dask/dataframe/core.py

```diff
@@ -8034,31 +8035,51 @@ def maybe_shift_divisions(df, periods, freq):
 
 @wraps(pd.to_datetime)
 def to_datetime(arg, meta=None, **kwargs):
-    tz_kwarg = {"tz": "utc"} if kwargs.get("utc") else {}
+    if PANDAS_GT_200 and "infer_datetime_format" in kwargs:
+        warnings.warn(
+            "The argument 'infer_datetime_format' is deprecated and will be removed in a future version. "
+            "A strict version of it is now the default, see "
+            "https://pandas.pydata.org/pdeps/0004-consistent-to-datetime-parsing.html. "
+            "You can safely remove this argument.",
+            UserWarning,
+        )
+        kwargs.pop("infer_datetime_format")
+
     if meta is None:
         if isinstance(arg, Index):
-            meta = pd.DatetimeIndex([], **tz_kwarg)
+            meta = get_meta_library(arg).DatetimeIndex(
+                [], **{"tz": "utc"} if kwargs.get("utc") else {}
+            )
             meta.name = arg.name
         elif not (is_dataframe_like(arg) or is_series_like(arg)):
             raise NotImplementedError(
                 "dask.dataframe.to_datetime does not support "
                 "non-index-able arguments (like scalars)"
             )
         else:
-            meta = meta_series_constructor(arg)([pd.Timestamp("2000", **tz_kwarg)])
+            if not isinstance(arg, _Frame):
+                arg_meta = arg.head(2)
+            else:
+                if is_series_like(arg):
+                    arg_meta = (
+                        meta_series_constructor(arg)(["2000/1/1"])
+                        if arg.dtype.kind == "O"
+                        else arg._meta_nonempty
+                    )
+                else:
+                    arg_meta = arg._meta_nonempty
+                    assign_dict = dict()
+                    for col in arg_meta.columns:
+                        if arg_meta[col].dtype.kind == "O":
+                            assign_dict[col] = "2000" if col == "year" else "1"
+                        elif col == "year" and is_numeric_dtype(arg_meta[col]):
+                            assign_dict[col] = 2000
+                    arg_meta = arg_meta.assign(**assign_dict)
```

Comment:
This looks complicated -- why is this change needed?

Reply:
`cudf.to_datetime` has a somewhat nuanced handling of datetime precision compared to pandas that made it so the only way to get the correct meta here was by directly calling `xd.to_datetime` on some subset of the data. This gets a little hairy when using `_meta_nonempty`, which generates data that would error in `to_datetime` (string columns containing non-integers, int columns containing out of range years, etc.), which this code attempts to resolve.

As I'm writing this, I wonder if maybe a simpler solution here would be to disable dtype checking on the Dask object in the tests? Even with this code, there are still some cases I can think of where the dtypes of the meta and the computed frame wouldn't match up (at this point, we don't know if any of the columns contain timezone-aware dates), and I'd imagine we'd prefer keeping the original meta construction which, though inaccurate in a few cases, isn't likely to introduce a breaking change.

Do you have a preference here?

Reply:
Ended up going back to the original meta construction and loosening dtype checks on GPU tests - do we think it's worth emitting a warning here if we run into that case?